src/ec/eddsa.rs

```diff
@@ -131,12 +131,7 @@ impl<'a> Ed25519KeyPair {
                              core::ptr::null(), 0,
                              core::ptr::null(), 0);
         unsafe { GFp_x25519_sc_reduce(nonce.as_mut_ptr()) };
-        let mut r = Point {
-            x: [0; ELEM_LIMBS],
-            y: [0; ELEM_LIMBS],
-            z: [0; ELEM_LIMBS],
-            t: [0; ELEM_LIMBS],
-        };
+        let mut r = Point::new();
```

Comment:
Here is `r` at infinity? If so, I think `new_at_infinity()` or `at_infinity()` is a better name than `new()`. Note that the code in src/ec/suite_b/ops.rs defines `Point::new_at_infinity()` and it's good to be consistent with that. (You can imagine in the future we're likely to refactor things so that the P-{256, 384} code and the Ed25519 code share some of these data types.)